tests/test_elasticbeanstalk.py

```diff
@@ -100,3 +112,140 @@ def test_eb_env_terminate(self):
+                 'environment/re-jenkins/%s' % envname
+        factory = self.replay_flight_data(
+            'test_elasticbeanstalk_env_tag'
+        )
+        p = self.load_policy(
+            {
+                'name': 'eb-tag',
+                'resource': 'elasticbeanstalk-environment',
+                'filters': [
+                    {
+                        'type': 'value',
+                        'key': 'CNAME',
+                        'op': 'regex',
+                        'value': '.*inactive.*',
+                    }
+                ],
+                'actions': [
+                    {
+                        'type': 'tag',
+                        'key': 'tagTestKey',
+                        'value': 'tagTestValue'
+                    }
+                ]
+            },
+            session_factory=factory
+        )
+        with mock_datetime_now(date_parse('2017-11-10'), datetime):
+            resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['EnvironmentName'], envname)
+        # Note that changing tags on an EB Env performs an update to the
+        # environment, which can take 60+ seconds. Uncomment sleeps to
+        # record test data.
+        # sleep(5)
+        while self.query_env_status(factory(), envname) != 'Ready':
+            # sleep(30)
+            pass
+        self.assertEqual(
+            self.env_tags_dict(factory(), envarn).get('tagTestKey'),
+            'tagTestValue'
+        )
+
+    def test_unmark(self):
+        envname = 'c7n-eb-tag-test-inactive'
+        envarn = 'arn:aws:elasticbeanstalk:us-east-1:012345678901:' \
+                 'environment/re-jenkins/%s' % envname
+        factory = self.replay_flight_data(
+            'test_elasticbeanstalk_env_unmark'
+        )
+        p = self.load_policy(
+            {
+                'name': 'eb-tag',
+                'resource': 'elasticbeanstalk-environment',
+                'filters': [
+                    {
+                        'type': 'value',
+                        'key': 'CNAME',
+                        'op': 'regex',
+                        'value': '.*inactive.*',
+                    }
+                ],
+                'actions': [
+                    {
+                        'type': 'remove-tag',
+                        'tags': ['tagTestKey']
+                    }
+                ]
+            },
+            session_factory=factory
+        )
+        with mock_datetime_now(date_parse('2017-11-10'), datetime):
+            resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['EnvironmentName'], envname)
+        # Note that changing tags on an EB Env performs an update to the
+        # environment, which can take 60+ seconds. Uncomment sleeps to
+        # record test data.
+        # sleep(5)
+        while self.query_env_status(factory(), envname) != 'Ready':
```

Comment:
can you manually delete some of the intermediate waiting recorded json, it cuts down on the recorded json here quite a bit I suspect, it requires manually deleting the corresponding describe jsons, and reordering the remaining files.